core/services/chainlink/config_general_secrets.go

```diff
@@ -32,3 +38,29 @@ func (g *generalConfig) PyroscopeAuthToken() string {
 	}
 	return string(*g.secrets.Pyroscope.AuthToken)
 }
+
+func (g *generalConfig) MercuryCredentials(url string) (username, password string, err error) {
+	if g.secrets.Mercury.Credentials == nil {
+		return "", "", errors.New("no Mercury credentials were specified in the config")
+	}
+	for _, creds := range g.secrets.Mercury.Credentials {
```

Comment:
curious on why this is a list of credentials instead of a single credential pair... if load balancing will happen on the infra level, how would this implementation affect that?

Reply:
Load balancing should not affect. It's only one set of credentials per server URL.